src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/poptimizer/costmodel/CFRNumberOfTermsShippedInRequests.java

```diff
@@ -33,20 +33,17 @@ public CompletableFuture<Integer> initiateCostEstimation( final PhysicalPlan pla
 		final int numberOfJoinVars;
 		final CompletableFuture<Integer> futureIntResSize;
 
-		if ( lop instanceof UnaryLogicalOp ) {
+		if ( lop instanceof LogicalOpTPAdd || lop instanceof LogicalOpBGPAdd || lop instanceof LogicalOpGPAdd  ) {
 			SPARQLGraphPattern pattern;
 			if (lop instanceof LogicalOpTPAdd) {
 				pattern = ((LogicalOpTPAdd) lop).getTP();
 
 			} else if (lop instanceof LogicalOpBGPAdd) {
 				pattern = ((LogicalOpBGPAdd) lop).getBGP();
 
-			} else if (lop instanceof LogicalOpGPAdd) {
+			} else {
 				pattern = ((LogicalOpGPAdd) lop).getPattern();
 			}
-			else {
-				throw createIllegalArgumentException(lop);
-			}
```

Comment:
Please revert back to the original version of this part -- just as an additional safety measure in case we change something here later.